react/ProductListContext.tsx

```diff
@@ -1,27 +1,45 @@
 import React, { createContext, useContext, useReducer, FC } from 'react'
 
-interface State {
-  visibleProducts: any[]
+const DEFAULT_STATE = {
```

Comment:
```suggestion
const DEFAULT_STATE: State = {
```

And them you can remove the `nextImpressions: [] as Product[]`